lisp/ess-tracebug.el

```diff
@@ -2355,50 +2445,12 @@ Arguments IGNORE and NOCONFIRM currently not used."
   (ess-watch)
   (message "Watch reverted"))
 
-(defvar ess-watch-mode-map nil
-  "Keymap for the *R watch* buffer.
-
-\\{ess-watch-mode-map}
-")
-
-(unless ess-watch-mode-map
-  (setq ess-watch-mode-map (make-sparse-keymap))
-  (when (boundp 'special-mode-map)
-    (set-keymap-parent ess-watch-mode-map special-mode-map))
-  (define-key ess-watch-mode-map "?" 'ess-watch-help)
-  (define-key ess-watch-mode-map "k" 'ess-watch-kill)
-  ;; (define-key ess-watch-mode-map "u" 'ess-watch-undelete)
-  ;; editing requires a little more work.
-  (define-key ess-watch-mode-map "a" 'ess-watch-add)
-  (define-key ess-watch-mode-map "i" 'ess-watch-insert)
-  (define-key ess-watch-mode-map "e" 'ess-watch-edit-expression)
-  (define-key ess-watch-mode-map "r" 'ess-watch-rename)
-  (define-key ess-watch-mode-map "q" 'ess-watch-quit)
-  (define-key ess-watch-mode-map "u" 'ess-watch-move-up)
-  (define-key ess-watch-mode-map "U" 'ess-watch-move-down)
-  (define-key ess-watch-mode-map "d" 'ess-watch-move-down)
-  (define-key ess-watch-mode-map "n" 'ess-watch-next-block)
-  (define-key ess-watch-mode-map "p" 'ess-watch-previous-block)
-  ;; R mode keybindings.
-  (define-key ess-watch-mode-map "\C-c\C-s" 'ess-watch-switch-process)
-  (define-key ess-watch-mode-map "\C-c\C-y" 'ess-switch-to-ESS)
-  (define-key ess-watch-mode-map "\C-c\C-z" 'ess-switch-to-end-of-ESS)
-  ;; Debug keys:
-  )
-
 
 (defface ess-watch-current-block-face
   '((default (:inherit highlight)))
   "Face used to highlight current watch block."
   :group 'ess-debug)
 
-(defvar  ess-watch-current-block-overlay nil
-  "The overlay for currently selected block in the R watch buffer .")
-(make-variable-buffer-local 'ess-watch-current-block-overlay)
-
-(defvar ess-watch-buffer "*R watch*"
-  "Name of the watch buffer.")
-
 (defvar  ess-watch-start-block "@----"  ;; fixme: make defcustom and modify the injected command correspondingly
```

Comment:
Why is this one not moved?

Reply:
It is - it's on line 166 now. Or have I misunderstood you?

Reply:
I meant that you moved all other defvars around it but didn't move this one. Nevermind.

Reply:
Oh, I see. You're talking about `ess-watch-start-block`. I only moved the variables that the byte compiler was complaining about in order to make as few changes as possible. But if you want I can move this up with the others.

Reply:
It's fine. We will have to reorganize that code anyhow at some later stage.

Reply:
OK, I think this is ready to be merged then. Let me know if I missed something.